swap_meet/vendor.py

```diff
@@ -1,2 +1,73 @@
 class Vendor:
-    pass
\ No newline at end of file
+    def __init__(self, inventory=None):
+        self.inventory = inventory if inventory is not None else []
+
+    def add(self, item):
+        self.inventory.append(item)
+        return item
+    
+    def remove(self, item):
+        try:
+            self.inventory.remove(item)
+            return item
+        except ValueError:
+            return False
+
+    def get_by_id(self, id):
+
+        for item in self.inventory:
+            if item.id == id:
+                return item
+        return None
+
+
+    def swap_items(self, other_vendor, my_item, their_item):
+        if not self.remove(my_item):
+            return False
+
+        if not other_vendor.remove(their_item):
+            self.add(my_item)
+            return False
```

Comment:
I'm curious as to why `line 29` is present. If either vendor's inventory doesn't contain their respective items we should abort the swap and just return `False`. Perhaps it was a mistake? 
```suggestion
        if not self.remove(my_item) or  not other_vendor.remove(their_item):
            return False
```

Reply:
Another thing that crosses my mind is the readability of your `if` statements. They have the responsibility of checking if an item is in the inventory as well as removing it if its there. I would actually prefer if we separated these two concerns into their own lines to more readily communicate their difference responsibilities and follow SRP closer.

Reply:
Your suggestion doesn't cover an edge case: if item exists in current class it will be deleted, but if it doesn't exist in other vendor instance, in this case we don't have an atomic operation since we already removed item from current class and this is the reason why we added line 29 to rollback changes.